ui/frontend/index.tsx

```diff
@@ -24,6 +24,13 @@ import PageSwitcher from './PageSwitcher';
 import playgroundApp from './reducers';
 import Router from './Router';
 import configureStore from './configureStore';
+import openWebSocket from './websocket';
+
+const params = new URLSearchParams(window.location.search);
+if (params.has('websocket') && params.get('websocket') === '1') {
+  // Might be null;
+  const socket = openWebSocket();
```

Comment:
Do we need to worry about the JS runtime garbage-collecting this `socket` variable? If that happens, our metrics will be skewed to show people disconnecting even though they are still on the page.

Reply:
Not sure about whether open sockets can be garbage collected. 
I'll move the WebSocket to top-level since it's going to be used by other components anyway.